test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -399,6 +466,2090 @@ static public void testWithFieldOnNonExistentClass() throws Throwable {
+		withFloat = generateWither(valueFloatClass, "f", float.class);
+
+		float f = Float.MAX_VALUE;
+		float fNew = Float.MIN_VALUE;
+		Object valueFloat = makeValueFloat.invoke(f);
+
+		assertEquals(getFloat.invoke(valueFloat), f);
+
+		valueFloat = withFloat.invoke(valueFloat, fNew);
+		assertEquals(getFloat.invoke(valueFloat), fNew);
+	}
+
+	/*
+	 * Create a value type with a reference member
+	 * 
+	 * value ValueObject {
+	 * 	Object val;
+	 * }
+	 */
+	@Test(priority = 1)
+	static public void testCreateValueObject() throws Throwable {
+		String fields[] = { "val:Ljava/lang/Object;:value" };
+
+		valueObjectClass = ValueTypeGenerator.generateValueClass("ValueObject", fields);
+
+		makeValueObject = lookup.findStatic(valueObjectClass, "makeValue",
+				MethodType.methodType(valueObjectClass, Object.class));
+
+		Object val = (Object)0xEEFFEEFF;
+		Object valNew = (Object)0xFFEEFFEE;
+
+		getObject = generateGetter(valueObjectClass, "val", Object.class);
+		withObject = generateWither(valueObjectClass, "val", Object.class);
+
+		Object valueObject = makeValueObject.invoke(val);
+
+		assertEquals(getObject.invoke(valueObject), val);
+
+		valueObject = withObject.invoke(valueObject, valNew);
+		assertEquals(getObject.invoke(valueObject), valNew);
+	}
+
+	/*
+	 * Create an assorted value type with long alignment
+	 * 
+	 * value AssortedValueWithLongAlignment {
+	 *	flattened Point2D point;
+	 *	flattened Line2D line;
+	 *	flattened ValueObject o;
+	 *	flattened ValueLong l;
+	 *	flattened ValueDouble d;
+	 *	flattened ValueInt i;
+	 *	flattened Triangle2D tri;
+	 * }
+	 */
+	@Test(priority = 4)
+	static public void testCreateAssortedValueWithLongAlignment() throws Throwable {
+		String fields[] = {
+				"point:QPoint2D;:value",
+				"line:QFlattenedLine2D;:value",
+				"o:QValueObject;:value",
+				"l:QValueLong;:value",
+				"d:QValueDouble;:value",
+				"i:QValueInt;:value",
+				"tri:QTriangle2D;:value" };
+		Class assortedValueClass = ValueTypeGenerator.generateValueClass("AssortedValueWithLongAlignment", fields);
+
+		MethodHandle makeAssorted = lookup.findStatic(assortedValueClass, "makeValueGeneric",
+				MethodType.methodType(assortedValueClass, Object.class, Object.class, Object.class, Object.class,
+						Object.class, Object.class, Object.class));
+
+		/*
+		 * Getters are created in array getterAndWither[i][0] according to the order of fields i
+		 * Withers are created in array getterAndWither[i][1] according to the order of fields i
+		 */
+		MethodHandle[][] getterAndWither = new MethodHandle[fields.length][2];
+		for (int i = 0; i < fields.length; i++) {
+			String field = (fields[i].split(":"))[0];
+			getterAndWither[i][0] = generateGenericGetter(assortedValueClass, field);
+			getterAndWither[i][1] = generateGenericWither(assortedValueClass, field);
+		}
+
+		/*
+		 * create field objects
+		 */
+
+		/*
+		 * point
+		 */
+		int x = 0xFFEEFFEE;
+		int y = 0xAABBAABB;
+		/*
+		 * line
+		 */
+		int stX = 0xFFEEFFEE;
```

Comment:
put the code to create lines and triangles in a helper function, so you dont have to repeat this all the time